tests/ctest/test_elektra.c

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include <tests_internal.h>
+#include <elektra.h>
+
+static void set (const char * parentKeyName, const char * name, const char * value) {
+    // Open
+    KeySet * config = ksNew (0, KS_END);
+    Key * parentKey = keyNew (parentKeyName, KEY_END);
+    KDB * handle = kdbOpen (parentKey);
+    kdbGet (handle, config, parentKey);
+
+    printf ("Error occurred: %s\n", keyString (keyGetMeta (parentKey, "error/description")));
+
+    // Set
+    Key *key = keyNew(parentKeyName, KEY_END);
+    keyAddName(key, name);
+    keySetString(key, value);
+    ksAppendKey(config, key);
+
+    // Check
+    Key *checkKey = ksLookupByName (config, name, 0);
+    printf ("SET: value is %s\n", keyString (checkKey));
+
+    // Save
+    int result = kdbSet (handle, config, parentKey);
+    printf ("SET: result is %d\n", result);
+    if (result == -1) {
+        Key * problemKey = ksCurrent (config);
+        printf ("SET: problemKey is %s\n", keyName (problemKey));
+    }
+
+    // Close
+    kdbClose (handle, parentKey);
+    keyDel (parentKey);
+}
+
+static void check (const char * parentKeyName, const char * name) {
+    // Open
+    KeySet * config = ksNew (0, KS_END);
+    Key * parentKey = keyNew (parentKeyName, KEY_END);
+    KDB * handle = kdbOpen (parentKey);
+    kdbGet (handle, config, parentKey);
+
+    // Check
+    Key *checkKey = ksLookupByName (config, name, 0);
+    printf ("CHECK: value is %s\n", keyString(checkKey));
+
+    // Close
+    kdbClose (handle, parentKey);
+    keyDel (parentKey);
+}
+
+static void test_development ()
+{
+    const char * parentKey = "user/sw/elektra/kdb/#0/current";
```

Comment:
Btw. the test cases redirect `user` to `/tmp`, so you cannot use `kdb get` to check the key afterwards.